libgnucash/engine/gnc-budget.cpp

```diff
@@ -65,6 +65,12 @@ struct PeriodData
     std::string note;
     bool value_is_set;
     gnc_numeric value;
+
+    ~PeriodData()
+    {
+        note.clear ();
+        note.shrink_to_fit ();
+    };
```

Comment:
This is harmless but unnecessary. When the `PeriodData` is destroyed `note` will be too. std::string is responsible for its character memory, you don't need to do anything.